internal/cliutils/checkforupdate.go

```diff
@@ -0,0 +1,83 @@
+package cliutils
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strconv"
+
+	"github.com/glasskube/glasskube/internal/config"
+)
+
+type ReleaseInfo struct {
+	Version string `json:"version"`
+}
+
+func versionIsGreater(version1, version2 string) bool {
+	r := regexp.MustCompile(`(\d+)`)
+	v1 := r.FindAllString(version1, -1)
+	v2 := r.FindAllString(version2, -1)
+
+	for i := 0; i < len(v1) && i < len(v2); i++ {
+		num1, _ := strconv.Atoi(v1[i])
+		num2, _ := strconv.Atoi(v2[i])
+
+		if num1 < num2 {
+			return true
+		} else if num1 > num2 {
+			return false
+		}
+	}
+
+	return len(v1) < len(v2)
+}
+
+func checkForUpdate() (bool, string, error) {
+	url := "https://glasskube.dev/release.json"
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to fetch release information: %v", err)
+	}
+
+	defer func() {
+		if closeErr := resp.Body.Close(); closeErr != nil {
+			err = fmt.Errorf("failed to close response body: %v", closeErr)
+		}
```

Comment:
If I understand this correctly, `err` is assigned here, but it is never used because `err` is never returned in this function. This could work if `err` was part of a [named result](https://go.dev/doc/effective_go#named-results).

In this case though I think it would be preferable to ignore the close error, since we only read from the stream.

To do this without triggering the linter, the result of `Close()` must be explicitly assigned to `_`.